Util/PasswordUpdater.php

```diff
@@ -40,7 +41,7 @@ public function hashPassword(UserInterface $user)
 
         $encoder = $this->encoderFactory->getEncoder($user);
 
-        if ($encoder instanceof BCryptPasswordEncoder || $encoder instanceof SelfSaltingEncoderInterface) {
+        if ($encoder instanceof NativePasswordEncoder || $encoder instanceof SelfSaltingEncoderInterface || $encoder instanceof BCryptPasswordEncoder) {
```

Comment:
no need to add NativePasswordEncoder. This is covered by the SelfSaltingEncoderInterface.

And the check for BCryptPasswordEncoder could be dropped once we drop support for Symfony versions which implement SelfSaltingEncoderInterface on it, as that's only a BC layer for older versions (I don't remember when the interface was added, but it was before 3.4).